code/parse/parselo.cpp

```diff
@@ -3419,7 +3419,7 @@ int split_str(const char *src, int max_pixel_w, int *n_chars, const char **p_str
 		buffer[buf_index] = 0;  // null terminate it
 
 		gr_get_string_size(&sw, NULL, buffer);
-		if (sw >= max_pixel_w) {
+		if (sw >= max_pixel_w || buf_index >= max_line_length) {
```

Comment:
Should this check for length-1?  You have to account for not only the number of bytes but also the null terminator.

Reply:
Hence I check >= and not >.
If, with the char already added, the length of the string is equal to the buffer size (and thus would fill the buffer exactly with no \0), it will be rolled back until the last separator (or by one).
Should work like this

Reply:
Ok